Wrappers/Python/cil/framework/framework.py

```diff
@@ -2481,7 +2481,51 @@ def maximum(self, x2, *args, **kwargs):
     def minimum(self,x2, out=None, *args, **kwargs):
         return self.pixel_wise_binary(numpy.minimum, x2=x2, out=out, *args, **kwargs)
 
+
+    def sapyb(self, a, y, b, out=None, dtype=numpy.float32, num_threads=NUM_THREADS):
+        '''performs a*self + b * y. It is inline safe
+        
+        Parameters
+        ----------
+        a : multiplier for self, can be a number or a numpy array or a DataContainer
+        y : DataContainer 
+        b : multiplier for y, can be a number or a numpy array or a DataContainer
+        out : return DataContainer, if None a new DataContainer is returned, default None. 
+            out can be self or y.
+        dtype : forces the output to the type, default numpy float32
+        num_threads : number of threads to use during the calculation, using the CIL C library
+        
+        It will try to use the CIL C library and default to numpy operations, in case the C library does
+        not handle the types.
+        '''
+        do_numpy = False
```

Comment:
I think is better `do_c = True` and for `False` make it agnostic. Moreover, a `numba` implementation will be useful because it will work for `numpy`, `cupy` and `dask`

Reply:
Maybe `do_c` is clearer.

Reply:
updated using `use_c_lib` and added example.